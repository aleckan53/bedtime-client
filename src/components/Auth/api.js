import config from 'config'

export default {
  login (data) {
    return fetch(`${config.API_ENDPOINT}/login`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'content-type': 'application/json'
      }
    })
    .then(res => !res.ok ? res.json().then(e => Promise.reject(e)): res.json())
  },
  signup (data) {
    return fetch(`${config.API_ENDPOINT}/users`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'content-type': 'application/json'
      }
    })
    .then(res => !res.ok ? res.json().then(e => Promise.reject(e)) : res.json())
  }
} 