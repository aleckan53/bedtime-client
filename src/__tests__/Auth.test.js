import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Auth from '../components/Auth'
import { BrowserRouter } from 'react-router-dom'
import 'jest-fetch-mock'

describe(`Auth Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Auth />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('matches the snapshot', () => {
    const wrapper = shallow(<Auth />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('siwtches form to Signup', () => {
    const wrapper = shallow(<Auth />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()

  })
})