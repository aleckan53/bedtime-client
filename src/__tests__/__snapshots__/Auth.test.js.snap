// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Component matches the snapshot 1`] = `
<section
  className="Auth"
>
  <form
    onSubmit={[Function]}
  >
    <Login
      setState={[Function]}
      state={
        Object {
          "error": null,
          "formValid": false,
          "values": Object {},
        }
      }
    />
  </form>
  <div
    className="btnWrapper"
  >
    <button
      id="switch"
      onClick={[Function]}
      type="button"
    >
      Signup
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Continue as a guest
    </button>
  </div>
</section>
`;

exports[`Auth Component siwtches form to Signup 1`] = `
<section
  className="Auth"
>
  <form
    onSubmit={[Function]}
  >
    <Login
      setState={[Function]}
      state={
        Object {
          "error": null,
          "formValid": false,
          "values": Object {},
        }
      }
    />
  </form>
  <div
    className="btnWrapper"
  >
    <button
      id="switch"
      onClick={[Function]}
      type="button"
    >
      Signup
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Continue as a guest
    </button>
  </div>
</section>
`;

exports[`Auth Component siwtches form to Signup 2`] = `
<section
  className="Auth"
>
  <form
    onSubmit={[Function]}
  >
    <Signup
      setState={[Function]}
      state={
        Object {
          "error": null,
          "formValid": false,
          "values": Object {},
        }
      }
    />
  </form>
  <div
    className="btnWrapper"
  >
    <button
      id="switch"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Continue as a guest
    </button>
  </div>
</section>
`;
